bundles/org.eclipse.ui.views.log/src/org/eclipse/ui/internal/views/log/LogSession.java

```diff
@@ -42,11 +48,24 @@ public Date getDate() {
 		return date;
 	}
 
+	/**
+	 * Returns a pretty-print formatting for the date for this entry
+	 * 
+	 * @return the formatted date for this entry
+	 */
+	public String getFormattedDate() {
+		if (fDateString == null) {
+			fDateString = LOCAL_SDF.format(getDate().toInstant());
+		}
+		return fDateString;
+	}
+
 	public void setDate(String dateString) {
-		SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS"); //$NON-NLS-1$
 		try {
-			date = formatter.parse(dateString);
-		} catch (ParseException e) { // do nothing
+			LocalDateTime ldt = LocalDateTime.parse(dateString, LOCAL_SDF);
```

Comment:
`LogEntry` does

```
Date date = Date.from(Instant.from(GREGORIAN_SDF.parse(stringToParse)));
if (date != null) {
	fDate = date;
	fDateString = LOCAL_SDF.format(fDate.toInstant());
}
```
Not sure what the difference is exactly and why we need a gregorian_sdf also, but since LogEntry is the probably more mature implementation, mabye
- make `org.eclipse.ui.internal.views.log.LogEntry.GREGORIAN_SDF` and `org.eclipse.ui.internal.views.log.LogEntry.LOCAL_SDF` package-protected
- use the constant from over there and just use the same parsing logic here as well.

Reply:
Thanks @chrisrueger 👍 :)